packages/angular_devkit/build_angular/src/builders/browser-esbuild/index.ts

```diff
@@ -410,4 +369,87 @@ async function bundleGlobalStylesheets(
   return { outputFiles, initialFiles, errors, warnings };
 }
 
+/**
+ * Main execution function for the esbuild-based application builder.
+ * The options are compatible with the Webpack-based builder.
+ * @param initialOptions The browser builder options to use when setting up the application build
+ * @param context The Architect builder context object
+ * @returns An async iterable with the builder result output
+ */
+export async function* buildEsbuildBrowser(
+  initialOptions: BrowserBuilderOptions,
+  context: BuilderContext,
+): AsyncIterable<BuilderOutput> {
+  // Only AOT is currently supported
+  if (initialOptions.aot !== true) {
+    context.logger.error(
+      'JIT mode is currently not supported by this experimental builder. AOT mode must be used.',
+    );
+
+    return { success: false };
+  }
+
+  // Inform user of experimental status of builder and options
+  logExperimentalWarnings(initialOptions, context);
+
+  // Determine project name from builder context target
+  const projectName = context.target?.project;
+  if (!projectName) {
+    context.logger.error(`The 'browser-esbuild' builder requires a target to be specified.`);
+
+    return { success: false };
+  }
+
+  const normalizedOptions = await normalizeOptions(context, projectName, initialOptions);
+
+  // Clean output path if enabled
+  if (initialOptions.deleteOutputPath) {
+    deleteOutputDir(normalizedOptions.workspaceRoot, initialOptions.outputPath);
+  }
+
+  // Create output directory if needed
+  try {
+    await fs.mkdir(normalizedOptions.outputPath, { recursive: true });
+  } catch (e) {
+    assertIsError(e);
+    context.logger.error('Unable to create output directory: ' + e.message);
+
+    return { success: false };
+  }
+
+  // Initial build
+  yield await execute(initialOptions, normalizedOptions, context);
+
+  // Finish if watch mode is not enabled
+  if (!initialOptions.watch) {
+    return;
+  }
+
+  // Setup a watcher
+  const watcher = createWatcher({
+    polling: typeof initialOptions.poll === 'number',
+    interval: initialOptions.poll,
+    // Ignore the output path to avoid infinite rebuild cycles
+    ignored: [normalizedOptions.outputPath],
+  });
+
+  // Temporarily watch the entire workspace
+  watcher.add(context.workspaceRoot);
```

Comment:
Shouldn't this be the project root?

Reply:
It is temporarily overly broad right now but I wanted to ensure that any relevant changes are picked up initially.
My logic for workspace root was that it would capture package changes as well as tsconfig changes but the project root would work for essentially all other cases and avoid unrelated generated projects from trigger a rebuild.
This will become more fine-grained once the dependencies are propagated from the build steps either way.